app/src/main/java/org/oppia/app/player/audio/AudioFragment.kt

```diff
@@ -52,4 +62,25 @@ class AudioFragment : InjectableFragment() {
     languageCodeList.add("hi-en")
     return languageCodeList
   }
+
+  private fun showCellularDataDialogFragment() {
+    cellularDataInterface = object : CellularDataInterface {
+      override fun cellularDataUserPreference(doNotShowAlert: Boolean, playAudio: Boolean) {
+        // true, true -> save preference and play audio
+        // true, false -> save preference and do not show AudioFragment
+        // false, true -> do not save preference and play audio
+        // false, false -> do nothing and do not show AudioFragment
+      }
+    }
+
+    val previousFragment = fragmentManager?.findFragmentByTag(TAG_CELLULAR_DATA_DIALOG)
+    if (previousFragment != null) {
+      fragmentManager?.beginTransaction()?.remove(previousFragment)?.commitNow()
+    }
+    val dialogFragment = CellularDataDialogFragment.newInstance(
+      cellularDataInterface
+    )
+    // Might need to replace show() with showNow().
+    dialogFragment.show(fragmentManager, TAG_CELLULAR_DATA_DIALOG)
```

Comment:
We should also prevent this dialog from reshowing upon a configuration change since the user has already started playing audio, so the question is moot.

We may also need to retain this across states, but we should check with Chantel first on anticipated user behavior/expectations here. This second consideration shouldn't block this PR.

Reply:
Should I add this as a comment in this code and solve this once the discussion with chantel is done?

Reply:
It seems worthwhile to file the second part as an issue and add a TODO here (ie, whether we should ask for streaming on each state or one per exploration).

We should solve the configuration change case now, I think.

Reply:
Sounds great. will change this accordingly.